python/pyspark/pandas/indexes/base.py

```diff
@@ -2293,8 +2311,6 @@ def union(
         sdf_self = self._internal.spark_frame.select(self._internal.index_spark_columns)
         sdf_other = other_idx._internal.spark_frame.select(other_idx._internal.index_spark_columns)
         sdf = sdf_self.union(sdf_other.subtract(sdf_self))
```

Comment:
Could you try `sdf_self.union(sdf_other).exceptAll(sdf_self.intersectAll(sdf_other))`?

It seems working at least with the example above:

```py
>>> idx1 = ps.Index([1, 1, 2, 2, 2])
>>> idx2 = ps.Index([1, 1, 1, 2, 2])
>>>
>>> idx1.union(idx2)
Int64Index([1, 1, 1, 2, 2, 2], dtype='int64')
>>> idx1.to_pandas().union(idx2.to_pandas())
Int64Index([1, 1, 1, 2, 2, 2], dtype='int64')
```

I haven't tried the other examples and I'm not sure how expensive this is, though.

Reply:
Let me try. Thanks!